faster_whisper/transcribe.py

```diff
@@ -2085,9 +2085,11 @@ def restore_speech_timestamps(
             )
 
         else:
+            middle = (segment.start + segment.end) / 2
+            chunk_index = ts_map.get_chunk_index(middle)
```

Comment:
this is not necessarily correct, as you noticed it fails the tests as it produces wrong timestamps in the test file, you need to identify the reason for the wrong timestamps in the first place to decide whether the incorrect timestamps are caused by the pre/post processing or from the whisper model